### PR TITLE
use tao-core base classes to relation control

### PR DESCRIPTION
**Pull request by gabrielfs7**
- Use tao-core base classes for relation management
- Deprecate old class / methods

ATTENTION!!! Build breaks because classes on tao-core were not merged yet.

https://oat-sa.atlassian.net/browse/CON-12

### CODE REVIEW

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: model/relation/MediaRelationService.php [hunk 1]
```diff
@@ -23,16 +23,32 @@
...
-    public function getMediaRelations(FindAllQuery $query): MediaRelationCollection
+    /**
+     * @deprecated Use self::relations()
+     */
+    public function getMediaRelations(LegacyQuery $query): MediaRelationCollection
     {
         return $this->getMediaRelationRepository()->findAll($query);
     }
 
+    public function relations(FindAllQuery $query): ResourceRelationCollection
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/relation/MediaRelationService.php:42]: i think this name is a bit too generic

File: scripts/install/SetMediaManager.php [hunk 2]
```diff
@@ -79,6 +82,12 @@ public function __invoke($params)
 
         $this->getServiceManager()->register(FileSystemService::SERVICE_ID, $fsService);
 
+        /** @var ResourceRelationServiceInterface $resourceRelationService */
+        $resourceRelationService = $this->getServiceManager()->get(ResourceRelationServiceProxy::SERVICE_ID);
+        $resourceRelationService->addService('media', MediaRelationService::class);
+
+        $this->getServiceManager()->register(ResourceRelationServiceProxy::SERVICE_ID, $resourceRelationService);
```
Comment (mike-ionut-mihai-sandu-tao) [inline scripts/install/SetMediaManager.php:89]: should we add a verification for this ?

**Review by mike-ionut-mihai-sandu-tao — COMMENTED** [review]
lgtm, left a few comments

File: migrations/Version202010301232112234_taoMediaManager.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        /** @var ResourceRelationServiceInterface $resourceRelationService */
+        $resourceRelationService = $this->getServiceManager()->get(ResourceRelationServiceProxy::SERVICE_ID);
+        $resourceRelationService->addService('media', MediaRelationService::class);
+
+        $this->getServiceManager()->register(ResourceRelationServiceProxy::SERVICE_ID, $resourceRelationService);
+    }
+
+    public function down(Schema $schema): void
+    {
```
Comment (bartlomiejmarszal) [inline migrations/Version202010301232112234_taoMediaManager.php:30]: We should be able to remove services registered in the `up` method

File: scripts/install/SetMediaManager.php (hunk 2 above)
Comment (gabrielfs7) [reply scripts/install/SetMediaManager.php:89]: It is part of `tao-core` so we should rely on it IMO.

File: model/relation/MediaRelationService.php (hunk 1 above)
Comment (gabrielfs7) [reply model/relation/MediaRelationService.php:42]: It is following the interface. Which name to you suggest? `getRelations` ?

**Review by bartlomiejmarszal — COMMENTED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by bartlomiejmarszal — APPROVED** [review]
LGTM

File: migrations/Version202010301232112234_taoMediaManager.php (hunk 3 above)
Comment (gabrielfs7) [reply migrations/Version202010301232112234_taoMediaManager.php:30]: True. I have added new feature and test for this :)

File: model/relation/MediaRelationService.php (hunk 1 above)
Comment (gabrielfs7) [reply model/relation/MediaRelationService.php]: renamed as we discussed :)

**Pull request merged**